snapshots/devmapper/pool_device.go

```diff
@@ -75,6 +75,33 @@ func NewPoolDevice(ctx context.Context, config *Config) (*PoolDevice, error) {
 	return poolDevice, nil
 }
 
+func retry(ctx context.Context, f func() error) error {
+	var (
+		maxRetries = 100
+		retryDelay = 100 * time.Millisecond
+		retryErr   error
+	)
+
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		retryErr = f()
+		if retryErr == nil {
+			return nil
+		} else if retryErr != unix.EBUSY {
+			return retryErr
+		}
+
+		// Don't spam logs
+		if attempt%10 == 0 {
+			log.G(ctx).WithError(retryErr).Warnf("retrying... (%d of %d)", attempt, maxRetries)
```

Comment:
This is from the original code, but how about logging the total number of retries at the end? The current implementation could log  `retrying... (10 of 100)` even if retries 19 times.

Reply:
I don't know if there is any value in this information. BUSY related errors are usually hard to measure, and if it takes 19 attempts on one instance, doesn't mean that the same will apply on another one. This warning is more to convey that we're doing something more than 10 times and we'll likely be failing anyway.